### PR TITLE
Error handling

### LINKED ISSUE #22: Sad Path/Edge Case Testing
Add some tests for graphql's extensive error handling capabilities.

CreateFarmer with missing email/name.

Update/delete both farmer and grain without id.

### PR DESCRIPTION
**Pull request by cdelpone**
### Project Management

Co-contributors: @chazsimons @PhiMed 

Closes #22

------------------------------------------------------------------------------
### Feature Status
* [x] Completed

------------------------------------------------------------------------------
### Test coverage

* [x] 100% Model Testing
* [ ] 100% Feature Testing

If below 100%, please explain:
- Unsure how to test graphql fully

------------------------------------------------------------------------------
### Change Log

What I did:
- testing for sad paths, including error messaging 

What needs to be refactored:
- remove comment from find farmer by id in query types

### CODE REVIEW

**Review by HarlemSquirrel — COMMENTED** [review]
Really nice work here with GraphQL!

I think your tests on mutations and queries are great. I wouldn't go too crazy on testing edge cases there. 

You could test some more at the model level if you're worried about bad data. Even better would be to add more model validations and not null. For instance, I see Farmer only needs to have an email right now and Grain has no validations. We can avoid bad data by adding as many constraints as make sense such as email format and the necessary specs for grains. If something should always be present, add a `NOT NULL` to the database so it's also enforced at that level like you have for timestamps.

File: app/graphql/mutations/create_grain.rb [hunk 1]
```diff
@@ -7,14 +7,17 @@ class CreateGrain < BaseMutation
...
 
     type Types::GrainType
 
     def resolve(name: nil, moisture: nil, falling_number: nil, protein: nil, test_weight: nil, farmers_notes: nil, farmer_id:)
-
-      farmer = Farmer.where(id: farmer_id)
-      farmer.first.grains.create!(name: name, moisture: moisture, falling_number: falling_number, protein: protein, test_weight: test_weight, farmers_notes: farmers_notes, farmer_id: farmer_id)
+      if Farmer.exists?(farmer_id)
+        farmer = Farmer.find(farmer_id)
+        farmer.grains.create!(name: name, moisture: moisture, falling_number: falling_number, protein: protein, test_weight: test_weight, farmers_notes: farmers_notes, farmer_id: farmer_id)
```
[the remark this answers is not in the document]
Comment (chazsimons) [reply app/graphql/mutations/create_grain.rb:17]: I see that now. We can just test that the farmer with that id exists and create a Grain object with the farmer_id.

Makes sense.

[the remark this answers is not in the document]
Comment (HarlemSquirrel) [reply app/graphql/mutations/create_grain.rb]: If `farmer_id` is set as a foreign key in the database then the database might not let you save a value that doesn't point to an existing farmer.

**Comment by PhiMed** [comment]
Merging, thanks again for the feedback Kevin
